pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/rebalance/TableRebalancer.java

```diff
@@ -175,67 +166,127 @@ public RebalanceResult rebalance(TableConfig tableConfig, Configuration rebalanc
     if (currentAssignment.equals(targetAssignment)) {
       LOGGER.info("Table: {} is already balanced", tableNameWithType);
       if (reassignInstances) {
-        return new RebalanceResult(RebalanceResult.Status.DONE, "Instance reassigned, table is already balanced",
-            instancePartitionsMap, targetAssignment);
+        if (dryRun) {
+          return new RebalanceResult(RebalanceResult.Status.DONE,
+              "Instance reassigned in dry-run mode, table is already balanced", instancePartitionsMap,
```

Comment:
Can you elaborate on what instance re-assignment means? Maybe best to add it in the comments on RebalanceConfigConstants. We need to also doc it in along with the rebalance command. Let me know if there is a doc already, and I can read it.

Reply:
More comments can be found in the instance assignment related classes. The doc for instance assignment has not been added yet, will do in the following pr.